src/main/java/org/opentripplanner/routing/edgetype/factory/PatternHopFactory.java

```diff
@@ -171,12 +177,6 @@ public void run(Graph graph) {
             }
         }
 
-        /* Generate unique human-readable names for all the TableTripPatterns. */
-        TripPattern.generateUniqueNames(tripPatterns);
-
-        /* Generate unique short IDs for all the TableTripPatterns. */
-        TripPattern.generateUniqueIds(tripPatterns);
-
```

Comment:
I moved these 2 method above the loop witch iterates over `TripPattern`s. When I made `TripPattern` a `TransitEntity` the loop failed, because it insert `TripPattern` into a Set/Map, which calls the `hashCode()` method. Calling the `hashCode()` on a `TransiEntity` without the `id` deliberately fails - it may lead to unpredictable results.